File: app/lib/model/device.dart
/*
SPDX-FileCopyrightText: 2026 M5Stack Technology CO LTD
SPDX-License-Identifier: MIT
*/

import 'package:uuid/v4.dart';

class Device {
  String mac;
  String? name;
  int? uid;
  String? bindTime;

  Device({required this.mac, this.name, this.uid, this.bindTime});

  String getDisplayName() {
    if (name != null && name != "") {
      return name!;
    } else {
      return mac;
    }
  }

  Device.fromJson(Map<String, dynamic> json)
    : mac = json['mac'] ?? UuidV4().toString(),
      uid = json['uid'] as int?,
      name = json['name']?.toString(),
      bindTime = json["bind_time"]?.toString();

  Map<String, dynamic> toJson() {
    return {'mac': mac, 'name': name, 'uid': uid, 'bind_time': bindTime};
  }

  static List<Device> fromListJson(List<dynamic> list) {
    return list.map((i) => Device.fromJson(i as Map<String, dynamic>)).toList();
  }
}

File: server/internal/model/device.go
/*
SPDX-FileCopyrightText: 2026 M5Stack Technology CO LTD
SPDX-License-Identifier: MIT
*/

package model

type DeviceInfo struct {
	Mac      string `json:"mac" v:"required" description:"Mac address"`
	Name     string `json:"name" v:"required" description:"Name"`
	Uid      int64  `json:"uid"  orm:"uid"  description:"Bound user UID"`                // Bound user UID
	BindTime string `json:"bind_time" orm:"bind_time" description:"Device binding time"` // Device binding time
}

type IPLocation struct {
	Status      string  `json:"status"`
	Country     string  `json:"country"`
	CountryCode string  `json:"countryCode"`
	Region      string  `json:"region"`
	RegionName  string  `json:"regionName"`
	City        string  `json:"city"`
	Zip         string  `json:"zip"`
	Lat         float64 `json:"lat"`
	Lon         float64 `json:"lon"`
	TimeZone    string  `json:"timezone"`
	Isp         string  `json:"isp"`
	Org         string  `json:"org"`
	As          string  `json:"as"`
	Query       string  `json:"query"`
}
